src/RA_StringUtils.h

```diff
@@ -606,14 +632,81 @@ _NODISCARD _Success_(0 < return < strsz) inline auto __cdecl tcslen_s(_In_reads_
 /// <summary>
 /// Determines if <paramref name="sString" /> starts with <paramref name="sMatch" />.
 /// </summary>
-GSL_SUPPRESS_F6
-_NODISCARD bool StringStartsWith(_In_ const std::wstring& sString, _In_ const std::wstring& sMatch) noexcept;
+template<typename CharT, typename = std::enable_if_t<is_char_v<CharT>>>
+GSL_SUPPRESS_F6 _NODISCARD bool StringStartsWith(_In_ const std::basic_string<CharT>& sString,
+                                                 _In_ const std::basic_string<CharT>& sMatch) noexcept
+{
+    if (sMatch.length() > sString.length())
+        return false;
+
+    return (sString.compare(0, sMatch.length(), sMatch) == 0);
+}
+
+/// <summary>
+/// Determines if <paramref name="sString" /> starts with <paramref name="sMatch" />.
+/// </summary>
+template<typename CharT, typename = std::enable_if_t<is_char_v<CharT>>>
+GSL_SUPPRESS_F6 _NODISCARD bool StringStartsWith(_In_ const std::basic_string<CharT>& sString,
+                                                 _In_ const CharT* restrict sMatch) noexcept
+{
+    if (tcslen_s(sMatch) > sString.length())
```

Comment:
`tcslen_s` should be called once and stored in a variable instead of recalculated on line 655.

Reply:
done